### DISCUSSION TITLE
Sync specific LDAP user using API

### DISCUSSION THREAD

**Discussion by tacerus** [keycloak/keycloak#12208] [Q&A]
Hi!

In the API documentation I found the endpoints `/{realm}/user-storage/{id}/sync` and `/{realm}/user-storage/{parentId}/mappers/{id}/sync`, however they only seem to allow for synchronization of **all** federated users.

Is there a way to trigger a sync for a **specific** user?

This would be useful if, for example, a user or administrator uses a tool which updates data on the LDAP end - since they only update the data of a single user, it would not make sense for the tool to trigger a full sync of all users (using the endpoints mentioned before) if it's known only attributes on a single user changed.

Thanks for any ideas!

**Comment by ihsanciftci** [comment]
Also updating groups of this specific users according to applied group mapper.

Updating single user and its related groups will be handful for the login scenario. After a successful login, the user info and group info will be up-to-date.
